### DISCUSSION TITLE
没有办法直接可以在OPENWRT下去执行这个脚本

### DISCUSSION THREAD

**Discussion by loofeal** [XIU2/CloudflareSpeedTest#346] [讨论]
我试用了另一个开发者的脚本最后提取出来的速度并没有想象中那么快。你这边的比较快一些。但是希望你能推出一个脚本

https://raw.githubusercontent.com/badafans/better-cloudflare-ip/master/shell/cf-openwrt.sh

你看看 怎么去修改

**Comment by XIU2** [comment]
你直接下载对应版本的 CloudflareST 解压运行就行了呀（只要你会 Linux 基本操作就够了）。

```
# 如果是第一次使用，则建议创建新文件夹（后续更新请跳过该步骤）
mkdir CloudflareST

# 进入文件夹（后续更新，只需要从这里重复下面的命令即可）
cd CloudflareST

# 下载 CloudflareST 压缩包（自行根据需求替换 URL 中版本号和文件名）
wget -N https://github.com/XIU2/CloudflareSpeedTest/releases/download/v1.4.10/CloudflareST_linux_arm64.tar.gz

# 解压（不需要删除旧文件，会直接覆盖，自行根据需求替换 文件名）
tar -zxf CloudflareST_linux_arm64.tar.gz

# 赋予执行权限
chmod +x CloudflareST

# 运行
./CloudflareST
```

主要是需要你自己去判断一下到底要使用哪个版本的 CloudflareST，然后修改上面的下载地址即可。

系统 | 架构 | 32/64 | 文件选择 |
:---- | :---- | :---- | :----
**Linux** | ARM | 32位 | CloudflareST_linux_**arm**.tar.gz
**Linux** | ARM  | 64位 | CloudflareST_linux_**arm64**.tar.gz
**Linux** | Mips | 32位 | CloudflareST_linux_**mips**.tar.gz
**Linux** | Mips | 64位 | CloudflareST_linux_**mips64**.tar.gz
**Linux** | Mipsle | 32位 | CloudflareST_linux_**mipsle**.tar.gz
**Linux** | Mipsle | 64位 | CloudflareST_linux_**mips64le**.tar.gz

路由器一般是 **Mips、Mipsle 架构**，而软路由则要看是不是 **ARM 架构**，不清楚可以都试试，能正常运行就代表选对了。
